packages/web/src/components/tabbedPane.tsx

```diff
@@ -36,37 +36,164 @@ export const TabbedPane: React.FunctionComponent<{
+            visibleCount = index;
+            break;
+          }
+        }
+        return visibleCount;
+      }
+    }
+
+    return tabs.length;
+  }, []);
+
+  const adjustElementRenderings = React.useCallback(() => {
+    const container = containerRef.current;
+    if (!container)
+      return;
+
+    const containerWidth = container.getBoundingClientRect().width;
+    setContainerWidth(containerWidth);
+
+    const initialAvailableWidth = containerWidth - (overflowTabs.length > 0 ? tabbedPaneWidth : 0);
+    const finalAvailableWidth = containerWidth - tabbedPaneWidth;
+
+    const visibleCount = calculateVisibleTabCount(initialAvailableWidth, tabWidths, tabs) === tabs.length
+      ? tabs.length
+      : calculateVisibleTabCount(finalAvailableWidth, tabWidths, tabs);
+
+    const visibleTabsList = tabs.slice(0, visibleCount);
+    const overflowTabsList = tabs.slice(visibleCount);
+
+    setVisibleTabs(visibleTabsList);
+    setOverflowTabs(overflowTabsList);
+  }, [tabWidths, overflowTabs, tabbedPaneWidth, tabs, calculateVisibleTabCount]);
+
+  // Initial measurement and setup
+  React.useEffect(() => {
+    if (overflowMode !== 'select')
+      return;
+
+    measureContainerTabs();
+  }, [measureContainerTabs, overflowMode]);
+
+  // Adjust when Tab widths change
+  React.useEffect(() => {
+    if (overflowMode !== 'select')
+      return;
+
+    if (overflowTabs.length > 0)
+      adjustElementRenderings();
+  }, [adjustElementRenderings, overflowMode, overflowTabs.length]);
+
+  React.useEffect(() => {
+    if (overflowMode !== 'select')
+      return;
+
+    const container = containerRef.current;
+    if (!container)
+      return;
+
+    const handleResize = () => {
+      setTimeout(adjustElementRenderings, 0);
+    };
+
+    const resizeObserver = new ResizeObserver(handleResize);
+    resizeObserver.observe(container);
+    handleResize();
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, [adjustElementRenderings, overflowMode]);
+
+  return <div className='tabbed-pane' data-testid={dataTestId} ref={containerRef}>
     <div className='vbox'>
       <Toolbar>
         { leftToolbar && <div style={{ flex: 'none', display: 'flex', margin: '0 4px', alignItems: 'center' }}>
           {...leftToolbar}
         </div>}
-        {mode === 'default' && <div style={{ flex: 'auto', display: 'flex', height: '100%', overflow: 'hidden' }} role='tablist'>
-          {[...tabs.map(tab => (
-            <TabbedPaneTab
-              key={tab.id}
-              id={tab.id}
-              ariaControls={`${id}-${tab.id}`}
-              title={tab.title}
-              count={tab.count}
-              errorCount={tab.errorCount}
-              selected={selectedTab === tab.id}
-              onSelect={setSelectedTab}
-            />)),
-          ]}
+        {visibleTabs.length > 0 && <div style={{ flex: 'auto', display: 'flex', height: '100%', overflow: 'hidden' }} role='tablist'>
```

Comment:
Does `mode` continue to work in the same way?

Reply:
Yes. `mode` continues to work in the same way because by default every tab is set as visible tab here: https://github.com/microsoft/playwright/blob/23fa81312d3d74f87f5e29da156463c3278efdfb/packages/web/src/components/tabbedPane.tsx#L50-L51